lib/unison-prelude/src/Unison/Debug.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Unison.Debug (debug, debugM, whenDebug, debugLogM, DebugFlag (..)) where
+
+import Control.Applicative (empty)
+import Control.Monad (when)
+import Data.Set (Set)
+import qualified Data.Set as Set
+import qualified Data.Text as Text
+import Debug.Pretty.Simple (pTrace, pTraceM, pTraceShowId, pTraceShowM)
+import System.IO.Unsafe (unsafePerformIO)
+import UnliftIO.Environment (lookupEnv)
+
+data DebugFlag
```

Comment:
We can add more to this whenever needed.

Reply:
I wonder if this would he better as a string. Then it's much easier to add ad-hoc ones while debugging, and we also don't use up any symbols globally.